conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/data/DefaultDecoderNames.java

```diff
@@ -66,23 +73,90 @@
      * @param inputType the container type used as input for the encoder, 'LIST' or 'NONE'.
      * @param outType the container type used as output for the encoder, 'LIST', 'SET', 'OPTIONAL', or 'NONE'.
      */
-    static String getDefaultDecoderMethodName(TypeMirror type, ContainerType inputType, ContainerType outType) {
-        return getDefaultDecoderMethodName(getClassNameForTypeMirror(type), inputType, outType);
+    static Optional<CodeBlock> getDefaultDecoderFactory(
+            TypeMirror type, ContainerType inputType, ContainerType outType) {
+        return getClassNameForTypeMirror(type)
+                .filter(SUPPORTED_CLASSES::contains)
+                .map(className -> getDefaultDecoderMethodName(className, inputType, outType))
+                .map(decoderMethodName ->
+                        CodeBlock.of("$T.$L(runtime.plainSerDe())", ParamDecoders.class, decoderMethodName))
+                .or(() -> getComplexDecoderFactory(type, inputType, outType));
+    }
+
+    private static Optional<CodeBlock> getComplexDecoderFactory(
+            TypeMirror typeMirror, ContainerType inputType, ContainerType outType) {
+        return getUnknownDecoderFactoryFunction(typeMirror)
+                .map(factoryFunction -> CodeBlock.of(
+                        "$T.$L(runtime.plainSerDe(), $L)",
+                        ParamDecoders.class,
+                        getDefaultDecoderMethodName(Optional.empty(), inputType, outType),
+                        factoryFunction));
+    }
+
+    static Optional<CodeBlock> getUnknownDecoderFactoryFunction(TypeMirror typeMirror) {
+        // No need to handle int/double/boolean because they're covered by default handlers
+        switch (typeMirror.getKind()) {
+            case FLOAT:
+                return Optional.of(CodeBlock.of("$T::parseFloat", Float.class));
+            case LONG:
+                return Optional.of(CodeBlock.of("$T::parseLong", Long.class));
+            case BYTE:
+                return Optional.of(CodeBlock.of("$T::parseByte", Byte.class));
+            case SHORT:
+                return Optional.of(CodeBlock.of("$T::parseShort", Short.class));
```

Comment:
These could be added to the `ParamDecoder` class, which would need to implement a similar call to the 'complex' serializer. Unclear which is better since there's no drop-through directly to conjure.

Reply:
Think adding these here is fine because the factory logic is very simple. For more complex logic, I have a slight preference towards putting it explicitly in `ParamDecoders` as I find it more readable/testable.